libyul/Dialect.h

```diff
@@ -48,8 +48,12 @@ struct BuiltinFunction
 	ControlFlowSideEffects controlFlowSideEffects;
 	/// If true, this is the msize instruction.
 	bool isMSize = false;
-	/// If set, same length as the arguments, if true at index i, the i'th argument has to be a literal which means it can't be moved to variables.
-	std::optional<std::vector<bool>> literalArguments;
+	/// Must be empty or the same length as the arguments. If set at index i, the i'th argument has to be a literal which means it can't be moved to variables.
```

Comment:
```suggestion
	/// Must be empty or the same length as the arguments.
	/// If set at index i, the i'th argument has to be a literal of the given kind, which also means it can't be moved to variables.
```